### PR TITLE
Reduce some Flock randomness

### PR DESCRIPTION
**Pull request by FlameArrow57**
[GAMEMODES][BALANCE]

## About the PR 
This PR just reduces some randomness in Flock

Resource cubes at start now have a constant total amount that is the average of the previous amount, you always get 13 Flock floor tiles if possible, and sentinel chain target amount is no longer random and is reduced.

## Why's this needed? 
Reduces some randomness that can affect Flock games

Not sure if Sentinels really need to chain to 5 or 6 people

Also makes it easier to scale some of this with player count

### CODE REVIEW

**Comment by FlameArrow57** [comment]
Some randomness with the grip tool too. Can change if wanted, but need input

File: code/obj/flock/structure/rift.dm [hunk 1]
```diff
@@ -27,34 +27,35 @@
...
-					if (src.flock)
-						src.flock.claimTurf(flock_convert_turf(S))
-						if (sentinel_count > 0 && !flock_is_blocked_turf(S))
-							new /obj/flock_structure/sentinel(S, src.flock)
-							sentinel_count--
-					else
-						flock_convert_turf(S)
-					candidate_turfs -= S
-					break
+		for(var/turf/simulated/floor/feather/floor as anything in candidate_turfs)
```
Comment (TobleroneSwordfish) [inline code/obj/flock/structure/rift.dm:48]:
```suggestion
		for(var/turf/simulated/floor/floor as anything in candidate_turfs)
```

**Review by TobleroneSwordfish — APPROVED** [review]
lgtm, 5-6 chain targets was frankly way too high, I thought it was much lower

**Review by amylizzle — APPROVED** [review]
lgtm also

File: code/obj/flock/structure/rift.dm [hunk 2]
```diff
@@ -27,34 +27,35 @@
...
 		var/obj/flock_structure/egg/bit/bitegg = new(src.contents, src.flock)
 		eject += bitegg
 		var/list/candidate_turfs = list()
-		for(var/turf/simulated/floor/S in orange(src, 4))
+		for(var/turf/simulated/floor/S in range(src, 2)) // 5x5
+			if (istype(S, /turf/simulated/floor/feather))
+				continue
 			candidate_turfs += S
+		while (length(candidate_turfs) > 13)
+			candidate_turfs -= pick(candidate_turfs)
```
Comment (pali6) [inline code/obj/flock/structure/rift.dm:45]: Considering you already shuffle the turfs you could just move the shuffling here and truncate it to length 13 and get the same effect but faster and cleaner. But the numbers are so small that speed shouldn't matter much.

Comment (pali6) [inline code/obj/flock/structure/rift.dm:45]:
```suggestion
		shuffle_list(candidate_turfs)
		if (length(candidate_turfs) > 13)
			candidate_turfs = candidate_turfs.Copy(1, 14)
```
and then remove the other `shuffle_list` call.

**Review by pali6 — APPROVED** [review]

File: code/obj/flock/structure/rift.dm [hunk 3]
```diff
@@ -27,34 +27,35 @@
...
-					else
-						flock_convert_turf(S)
-					candidate_turfs -= S
-					break
+		for(var/turf/simulated/floor/floor as anything in candidate_turfs)
+			if (src.flock)
+				src.flock.claimTurf(flock_convert_turf(floor))
+			else
+				flock_convert_turf(floor)
+		shuffle_list(candidate_turfs)
```
Comment (pali6) [inline code/obj/flock/structure/rift.dm:53]: see previous comment
```suggestion
```

File: code/obj/flock/structure/rift.dm (hunk 2 above)
Comment (amylizzle) [reply code/obj/flock/structure/rift.dm:45]: that if(length) breaks it, you'd want something like `candidate_turfs = candidate_turfs.Copy(1, max(14, length(candidate_turfs)+1))` instead

Comment (pali6) [reply code/obj/flock/structure/rift.dm:45]: What? That would just mean that there's no effect.

Comment (amylizzle) [reply code/obj/flock/structure/rift.dm:45]: oh wait I'm an idiot, disregard

**Comment by pali6** [comment]
yell at me if I broke it somehow

**Pull request merged**